components/proxy/src/test/scala/com/hotels/styx/ServerConfigSchemaSpec.scala

```diff
@@ -0,0 +1,360 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx
+
+import java.util.Optional
+
+import com.hotels.styx.ServerConfigSchema.validateServerConfiguration
+import com.hotels.styx.infrastructure.configuration.ConfigurationParser
+import com.hotels.styx.infrastructure.configuration.ConfigurationSource.configSource
+import com.hotels.styx.infrastructure.configuration.yaml.YamlConfiguration
+import com.hotels.styx.infrastructure.configuration.yaml.YamlConfigurationFormat.YAML
+import org.scalatest.{FunSpec, ShouldMatchers}
+
+class ServerConfigSchemaSpec extends FunSpec with ShouldMatchers {
+
+  describe("Styx Server Configuration") {
+    it("Minimal configuration") {
+      validateServerConfiguration(yamlConfig(
+        """
+          |proxy:
+          |  connectors:
+          |    http:
+          |      port: 8080
+          |
+          |admin:
+          |  connectors:
+          |    http:
+          |      port: 9000
+          |
+          |services:
+          |  factories:
+          |    backendServiceRegistry:
+          |      class: "com.hotels.styx.proxy.backends.file.FileBackedBackendServicesRegistry$Factory"
+          |      config: {originsFile: "${originsFile:classpath:conf/origins.yml}"}
+        """.stripMargin)) should be(Optional.empty())
+    }
+
+    it("Proxy configuration is mandatory") {
+      validateServerConfiguration(yamlConfig(
+        """
+          |admin:
+          |  connectors:
+          |    http:
+          |      port: 9000
+          |
+          |services:
+          |  factories:
+          |    backendServiceRegistry:
+          |      class: "com.hotels.styx.proxy.backends.file.FileBackedBackendServicesRegistry$Factory"
+          |      config: {originsFile: "${originsFile:classpath:conf/origins.yml}"}
+        """.stripMargin)) should be(Optional.of("Missing a mandatory field 'proxy'"))
+    }
+
+    it("Admin configuration is mandatory") {
+      validateServerConfiguration(yamlConfig(
+        """
+          |proxy:
+          |  connectors:
+          |    http:
+          |      port: 8080
+          |
+          |services:
+          |  factories:
+          |    backendServiceRegistry:
+          |      class: "com.hotels.styx.proxy.backends.file.FileBackedBackendServicesRegistry$Factory"
+          |      config: {originsFile: "${originsFile:classpath:conf/origins.yml}"}
+        """.stripMargin)) should be(Optional.of("Missing a mandatory field 'admin'"))
+    }
+
+    it("Services configuration is mandatory") {
```

Comment:
The word `it` plus the string argument should form a sentence.

Reply:
Addressed this.